code/modules/virology/virology_goals.dm

```diff
@@ -0,0 +1,171 @@
+GLOBAL_LIST_EMPTY(virology_goals)
+
+/datum/virology_goal
+	var/name = "Generic Virology Goal"
+	var/delivered_amount = 0
+	var/delivery_goal = 15
+	var/completed = FALSE
+
+/datum/virology_goal/proc/get_report()
+	return "Complete this goal."
+
+/datum/virology_goal/proc/get_ui_report()
+	return "Complete this goal."
+
+/datum/virology_goal/proc/check_completion(list/datum/reagent/reagent_list)
+	return TRUE
+
+/datum/virology_goal/Destroy()
+	LAZYREMOVE(GLOB.virology_goals, src)
+	. = ..()
+
+/datum/virology_goal/Topic(href, href_list)
+	..()
+
+	if(!check_rights(R_EVENT))
+		return
+
+	if(href_list["remove"])
+		qdel(src)
+
+/datum/virology_goal/propertysymptom
+	name = "Symptom With Properties Viral Sample Request"
+	var/goal_symptom //Type path of the symptom
+	var/goal_symptom_name
+	var/goal_property
+	var/goal_property_text
+	var/goal_property_value
+
+/datum/virology_goal/propertysymptom/New()
+	var/type = pick(subtypesof(/datum/symptom))
+	var/datum/symptom/S = new type()
+	goal_symptom = S.type
+	goal_symptom_name = S.name
+	goal_property = pick("resistance", "stealth", "stage_rate", "transmittable")
+	if(goal_property == "stage_rate")
+		goal_property_text = "stage rate"
+	else
+		goal_property_text = goal_property
```

Comment:
Why not make the string `"stage rate"` and just do `goal_property_text = goal_property`

Reply:
the goal_property is the actual variable of the stage rate and goal_property_text is what shows up on the paper and TGUI